CI fails due to the reference of the internal _pytest.python_api.RaisesContext

### Issue Summary

* CI failing due to the inability to import `from _pytest.python_api import RaisesContext`

```bash
  tests/pyavd/workflows/test_verify_inputs.py:13: in <module>
      from _pytest.python_api import RaisesContext
  E   ImportError: cannot import name 'RaisesContext' from '_pytest.python_api' (/home/runner/work/avd/avd/python-avd/.tox/py312/lib/python3.12/site-packages/_pytest/python_api.py)
```

* internal `RaisesContext` was removed from pytest in 8.4.0 (released on 2025-06-02

https://github.com/pytest-dev/pytest/compare/8.3.5...8.4.0#diff-1118e2a1281a92eca6051881271a30ab8ada17354c867a086f68d9f2f7ea2807L979

### Which components of AVD are impacted

others

### AVD version

devel

### Ansible version

_No response_

### Python version

_No response_

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

```shell
run `tox`
```

### Relevant log output

```shell
tests/pyavd/workflows/test_verify_inputs.py:13: in <module>
      from _pytest.python_api import RaisesContext
  E   ImportError: cannot import name 'RaisesContext' from '_pytest.python_api' (/home/runner/work/avd/avd/python-avd/.tox/py312/lib/python3.12/site-packages/_pytest/python_api.py)
```

### Contributing Guide

- [x] I agree to follow this project Code of Conduct

--- a/python-avd/tests/pyavd/workflows/test_verify_inputs.py
+++ b/python-avd/tests/pyavd/workflows/test_verify_inputs.py
@@ -7,14 +7,16 @@ import cProfile
 import logging
 import re
 from collections.abc import Callable
+from contextlib import AbstractContextManager
 from contextlib import nullcontext as does_not_raise
 
 import pytest
-from _pytest.python_api import RaisesContext
 
 from pyavd._cv.client.exceptions import CVDuplicatedDevices
 from pyavd._cv.workflows.models import CVDevice
 from pyavd._cv.workflows.verify_inputs import identify_duplicated_devices, verify_device_inputs
+
+ExpectedExceptionContext = AbstractContextManager[pytest.ExceptionInfo | None]
 
 TWO_DUPED_SERIAL_PATTERNS = [
     "\\('Duplicated devices found in inventory.*\\{"
@@ -567,7 +569,7 @@ def test_verify_device_inputs(
     logs_qty: int,
     expected_logs_patterns: list[str],
     expected_exception_patterns: list[str],
-    expected_exception: RaisesContext | does_not_raise,
+    expected_exception: ExpectedExceptionContext,
     *,
     strict_system_mac_address: bool,
 ) -> None:

